### ISSUE TITLE
[They Vote For You/production] ArgumentError: SMTP To address may not be blank: []

### ISSUE THREAD

**Issue by mlandauer** [openaustralia/publicwhip#1344]
## Backtrace

    line 8 of [PROJECT_ROOT]/app/jobs/alert_watches_job.rb: block in perform
    line 7 of [PROJECT_ROOT]/app/jobs/alert_watches_job.rb: perform

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/37542/faults/84120508)

**Comment by mlandauer** [comment]
I'm guessing that this issue is the cause of #1342

**Comment by mlandauer** [comment]
The error in production is happening on running `AlertWatchesJob#perform` with policy `gid://publicwhip/Policy/114` and version `gid://publicwhip/PaperTrail::Version/3963`

**Comment by mlandauer** [comment]
This happened because of a couple of problems on top of each other:
* We used to have a validation on user to ensure that name was unique but there were duplicate entries in the database
* Then, I did a migration in a47c139371b51e7df27f69944adf20be1fb72d7d which didn't actually check properly that the update worked. Ugh. See https://github.com/openaustralia/publicwhip/commit/a47c139371b51e7df27f69944adf20be1fb72d7d#r67025395 - I should have used `update!` rather than `update`

So, for a small number of confirmed users (about 25) which used names that were not unique in the database the migration didn't work properly and their email and name was set to nil. Eek. I used a database snapshot from a few days ago to reinstate a copy of the old database and manually fill in the missing data from the backup. I had to ensure that devise didn't send out reconfirmation emails on doing that.

Then the problem was that the background job for sending out alerts for a single policy update kept failing (because one of the watches had an invalid email) but it kept resending them for all. I made a short term workaround to check the email first but this has now been removed after the data has been fixed. Ideally we would have a single background job per email sent to users so that we wouldn't ever have this happen to multiple people. The worst that could happen is that a job fails for the particular user that is being emailed. This is similar to how we do things on PlanningAlerts now. However, we are using sidekiq rather than delayed_job so this is more doable without it being incredibly slow.
